crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1466,22 +1468,51 @@ impl AuthorityStore {
             return Ok(());
         }
 
-        let mut total_storage_rebate = 0;
-        let mut total_sui = 0;
         info!("Starting SUI conservation check. This may take a while..");
         let cur_time = Instant::now();
+        let mut pending_objects = vec![];
         let mut count = 0;
-        for o in self.iter_live_object_set() {
-            match o {
-                LiveObject::Normal(object) => {
-                    total_storage_rebate += object.storage_rebate;
-                    // get_total_sui includes storage rebate, however all storage rebate is
-                    // also stored in the storage fund, so we need to subtract it here.
-                    total_sui += object.get_total_sui(self)? - object.storage_rebate;
-                    count += 1;
+        let package_cache = PackageObjectCache::new(self.clone());
```

Comment:
I did notice that the cache has no eviction--probably fine for realistic package counts today, but could be an OOM risk in the future.

Reply:
Fixed using an LRU cache